indico/modules/search/controllers.py

```diff
@@ -63,11 +63,17 @@ class InternalSearch(IndicoSearchProvider):
     def search(self, query, access, object_type=SearchTarget.event, page=1, params=None):
         # Without any search provider, internally we only support categories and events
         if object_type == SearchTarget.category:
-            total, results = InternalSearch.search_categories(page, query)
+            if params.get('category'):
+                total, results = InternalSearch.search_subcategories(page, query, params['category'])
+            else:
+                total, results = InternalSearch.search_categories(page, query)
         elif object_type == SearchTarget.event:
-            total, results = InternalSearch.search_events(page, query)
+            if params.get('category'):
+                total, results = InternalSearch.search_category_events(page, query, params['category'])
+            else:
+                total, results = InternalSearch.search_events(page, query)
         else:
-            return 0, [], []
+            return 0, 0, [], []
```

Comment:
Even if this commit ends up being discarded, keep in mind this line was returning an incorrect tuple @plourenco.

Reply:
Yes, `pages` was not supposed to be here, but it had to be added just recently. I'm updating this on the base branch and we'll rebase it.